### PR TITLE
fix go vet warnings

### LINKED ISSUE #5798: Resolve `go vet -mod=mod ./...` suggestions
**Describe the problem/challenge you have**
[A description of the current limitation/problem/challenge that you are experiencing.]
Resolve `go vet -mod=mod ./...` suggestions

```shell
❯ go vet -mod=mod ./...
# github.com/vmware-tanzu/velero/pkg/itemoperation
pkg/itemoperation/backup_operation.go:40:2: struct field tag `json:resourceIdentifier` not compatible with reflect.StructTag.Get: bad syntax for struct tag value
pkg/itemoperation/backup_operation.go:43:2: struct field tag `json:operationID` not compatible with reflect.StructTag.Get: bad syntax for struct tag value
pkg/itemoperation/restore_operation.go:40:2: struct field tag `json:resourceIdentifier` not compatible with reflect.StructTag.Get: bad syntax for struct tag value
pkg/itemoperation/restore_operation.go:43:2: struct field tag `json:operationID` not compatible with reflect.StructTag.Get: bad syntax for struct tag value
pkg/itemoperation/shared.go:43:2: struct field OperationUnits repeats json tag "nTotal" also at shared.go:39
# github.com/vmware-tanzu/velero/internal/hook
internal/hook/item_hook_handler_test.go:805:18: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:818:18: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:869:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:870:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:899:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:900:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:918:22: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:919:22: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:939:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:940:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:957:22: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:958:22: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:978:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:979:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:997:22: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:998:22: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1025:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1026:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1066:22: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1067:22: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1088:22: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1089:22: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1097:22: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1098:22: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1106:22: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1107:22: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1121:22: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1122:22: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1142:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1143:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1153:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1154:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1164:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1165:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1177:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/item_hook_handler_test.go:1178:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:105:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:106:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:118:18: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:174:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:175:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:187:18: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:243:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:244:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:256:18: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:312:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:313:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:325:18: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:447:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:478:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:881:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:882:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:887:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:900:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:905:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:912:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:917:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:930:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:931:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:936:21: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
internal/hook/wait_exec_hook_handler_test.go:741:10: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
# github.com/vmware-tanzu/velero/pkg/cmd/cli/nodeagent
pkg/cmd/cli/nodeagent/server.go:114:2: the cancelFunc function is not used on all paths (possible context leak)
pkg/cmd/cli/nodeagent/server.go:118:3: this return statement may be reached without using the cancelFunc var defined on line 114
# github.com/vmware-tanzu/velero/pkg/persistence/mocks
pkg/persistence/mocks/backup_store.go:308:2: unreachable code
pkg/persistence/mocks/backup_store.go:313:2: unreachable code
pkg/persistence/mocks/backup_store.go:318:2: unreachable code
pkg/persistence/mocks/backup_store.go:323:2: unreachable code
pkg/persistence/mocks/backup_store.go:328:2: unreachable code
# github.com/vmware-tanzu/velero/pkg/repository
pkg/repository/manager.go:250:47: k8s.io/apimachinery/pkg/types.NamespacedName composite literal uses unkeyed fields
# github.com/vmware-tanzu/velero/pkg/controller
pkg/controller/backup_controller_test.go:743:28: k8s.io/apimachinery/pkg/apis/meta/v1.Time composite literal uses unkeyed fields
pkg/controller/backup_repository_controller_test.go:57:26: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
pkg/controller/backup_repository_controller_test.go:144:28: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
pkg/controller/backup_repository_controller_test.go:157:28: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
pkg/controller/backup_repository_controller_test.go:170:28: k8s.io/apimachinery/pkg/apis/meta/v1.Duration composite literal uses unkeyed fields
pkg/controller/restore_controller_test.go:659:1: TestvalidateAndCompleteWhenScheduleNameSpecified has malformed name: first letter after 'Test' must not be lowercase
# github.com/vmware-tanzu/velero/test/e2e/util/k8s
test/e2e/util/k8s/namespace.go:70:20: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
# github.com/vmware-tanzu/velero/test/e2e/util/velero
test/e2e/util/velero/velero_utils.go:1116:4: github.com/vmware-tanzu/velero/test/e2e.VeleroCLI2Version composite literal uses unkeyed fields
# github.com/vmware-tanzu/velero/test/e2e/privilegesmgmt
test/e2e/privilegesmgmt/ssr.go:60:8: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
# github.com/vmware-tanzu/velero/test/e2e/pv-backup
test/e2e/pv-backup/pv-backup-filter.go:34:9: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
test/e2e/pv-backup/pv-backup-filter.go:72:9: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
test/e2e/pv-backup/pv-backup-filter.go:135:9: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
# github.com/vmware-tanzu/velero/test/e2e/basic/resources-check
test/e2e/basic/resources-check/namespaces.go:95:9: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
test/e2e/basic/resources-check/namespaces.go:124:9: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
test/e2e/basic/resources-check/namespaces_annotation.go:70:9: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
test/e2e/basic/resources-check/rbac.go:86:9: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
# github.com/vmware-tanzu/velero/test/e2e/schedule
test/e2e/schedule/ordered_resources.go:139:9: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
test/e2e/schedule/schedule.go:55:9: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
# github.com/vmware-tanzu/velero/test/e2e/resource-filtering
test/e2e/resource-filtering/base.go:69:9: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
test/e2e/resource-filtering/exclude_label.go:79:9: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
test/e2e/resource-filtering/exclude_namespaces.go:112:9: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
test/e2e/resource-filtering/include_namespaces.go:108:9: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
test/e2e/resource-filtering/label_selector.go:78:9: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
# github.com/vmware-tanzu/velero/test/e2e/util/providers
test/e2e/util/providers/common.go:141:8: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
# github.com/vmware-tanzu/velero/test/e2e/bsl-mgmt
test/e2e/bsl-mgmt/deletion.go:140:20: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
# github.com/vmware-tanzu/velero/test/e2e/backups
test/e2e/backups/deletion.go:91:18: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
test/e2e/backups/sync_backups.go:50:9: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
test/e2e/backups/ttl.go:55:9: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
# github.com/vmware-tanzu/velero/test/e2e/basic
test/e2e/basic/namespace-mapping.go:33:20: github.com/vmware-tanzu/velero/test/e2e/util/kibishii.KibishiiData composite literal uses unkeyed fields
test/e2e/basic/namespace-mapping.go:82:9: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
test/e2e/basic/namespace-mapping.go:99:9: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
# github.com/vmware-tanzu/velero/test/e2e/upgrade
test/e2e/upgrade/upgrade.go:94:20: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
# github.com/vmware-tanzu/velero/test/e2e/migration
test/e2e/migration/migration.go:101:20: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak
```

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Velero version (use `velero version`): main
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

### PR DESCRIPTION
**Pull request by kaovilai**
- jsontag issues
- stop passing around and reassigning context without cancelling in e2e.
- cancel context if not used
- add vet checks to golangci.yaml

Signed-off-by: Tiger Kaovilai <tkaovila@redhat.com>

Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #5798

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

/kind changelog-not-required

### CODE REVIEW

File: hack/verify-vet.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
```
Comment (blackpiglet) [inline hack/verify-vet.sh:1]: Suggest removing this file and adding the govet linter in file `golangci.yaml`.

**Review by blackpiglet — APPROVED** [review]

File: golangci.yaml [hunk 2]
```diff
@@ -181,7 +181,7 @@ linters-settings:
         #     reason: "testing if blocked version constraint works."  # Reason why the version constraint exists. (Optional)
   govet:
     # report about shadowed variables
-    check-shadowing: true
+    # check-shadowing: true
```
Comment (reasonerjt) [inline golangci.yaml:184]: Could you please explain why this is commented?  Is it b/c it's always enabled by default or you want to turn it off?
In the doc it says:
```
Shadowed variables [¶](https://pkg.go.dev/github.com/golangci/govet#hdr-Shadowed_variables)
Flag: -shadow=false (experimental; must be set explicitly)

Variables that may have been unintentionally shadowed.
```

Comment (blackpiglet) [reply golangci.yaml:184]: This is my suggestion. 
`check-shadowing`'s default value is false, and when it's set to true, more warnings are reported. Most of them are false positives.

**Review by reasonerjt — APPROVED** [review]
Approving this one to avoid unnecessary rebase.

I do have some concerns around `# check-shadowing: true`, but we can fix it in a follow up change

**Pull request merged**